app/cdap/components/AddTransformation/SelectColumnsList/index.tsx

```diff
@@ -0,0 +1,247 @@
+      if (columnValue.length) {
+        setColumns(columnValue);
+      } else {
+        setColumns([]);
+      }
+    } else {
+      setColumns(columnData);
+    }
+  };
+
+  const handleFocus = () => {
+    ref?.current.focus();
+    setFocused(true);
+  };
+
+  return (
+    <section className={classes.columnsCountTextStyles}>
+      <div className={classes.selectColumnsHeaderStyles}>
+        <div>
+          {selectedColumnsCount
+            ? selectedColumnsCount > 10
+              ? selectedColumnsCount
+              : `0${selectedColumnsCount}`
+            : 'No '}{' '}
+          &nbsp;{T.translate('features.WranglerNewAddTransformation.columnsSelected')}
+        </div>
+        <div className={classes.searchFormControl}>
+          <input
+            className={focused ? classes.isFocused : classes.isBlurred}
+            onChange={handleSearch}
+            ref={ref}
+            onFocus={() => setFocused(true)}
+            onBlur={() => setFocused(false)}
+          />
+          <Box className={classes.searchInputAdornment} onClick={handleFocus}>
+            {SearchIcon()}
+          </Box>
+        </div>
+      </div>
+      {no_match.length === 0 ? (
+        <Box className={classes.noRecordWrapper}>
+          <Box className={classes.innerWrapper}>
+            {NoDataSVG}
+            <Typography className={classes.mainHeaderMessage}>
+              {T.translate('features.WranglerNewSelectCoulmnList.noColumns')}
+            </Typography>
+            <Typography className={classes.subHeaderMessage}>
+              {T.translate('features.WranglerNewSelectCoulmnList.noMatchColumnDatatype')}
+            </Typography>
+          </Box>
+        </Box>
+      ) : (
+        <TableContainer component={Box}>
+          <Table aria-label="recipe steps table" className={classes.tabledisplayStyles}>
+            <TableHead>
+              <TableRow className={`${classes.recipeStepsTableRowStyles} ${classes.rowsOfTable}`}>
+                <TableCell
+                  classes={{
+                    head: `${classes.recipeStepsTableHeadStyles} ${classes.columnstyles}`,
+                  }}
+                ></TableCell>
+                <TableCell
+                  classes={{
+                    head: `${classes.recipeStepsTableHeadStyles} ${classes.nullValueHead}`,
+                  }}
+                >
+                  {T.translate('features.WranglerNewAddTransformation.columns')}
+                </TableCell>
+                <TableCell
+                  classes={{
+                    head: `${classes.recipeStepsTableHeadStyles} ${classes.nullValueHead}`,
+                  }}
+                >
+                  {T.translate('features.WranglerNewAddTransformation.nullValues')}
+                </TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {columns.map((eachColumn, index) => {
+                if (directiveFunctionSupportedDataType.includes('all')) {
```

Comment:
Remove if else and use conditional statement

Reply:
add null checks as well

Reply:
Done